core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -741,6 +749,17 @@ public Command workerHeartbeat(long workerId, Map<String, Long> usedBytesOnTiers
     }
   }
 
+  private void processWorkerMetrics(String hostname, BlockHeartbeatTOptions options) {
+    if (options.getMetrics() == null || options.getMetricsSize() == 0) {
+      return;
+    }
+    List<Metric> metrics = new ArrayList<>();
+    for (alluxio.thrift.Metric metric : options.getMetrics()) {
+      metrics.add(Metric.from(metric));
+    }
+    mMetricsStore.putWorkerMetrics(MetricsSystem.WORKER_INSTANCE, hostname, metrics);
```

Comment:
Do you think it makes sense to send a delta instead of all metrics every heartbeat?